src/rule_definition_tools.jl

```diff
@@ -215,14 +219,22 @@ end
 "Declares properly hygenic inputs for propagation expressions"
 _propagator_inputs(n) = [esc(gensym(Symbol(:Δ, i))) for i in 1:n]
 
+"given the variable names, escaped but without types, makes setup expressions for projection operators"
+function _make_projectors(xs)
+    ps = map(x -> gensym(Symbol(:proj_, x.args[1])), xs)
```

Comment:
Do we  need to gensym this?
Or will macrohygine take case of it for us?

Reply:
I did think this got escaped whole somewhere on the path to the outside world. But will re-check.